src/components/SearchBar/SearchBar.tsx

```diff
@@ -1,42 +1,85 @@
 import * as React from 'react';
 
-import { Input, Button } from 'buildo-react-components';
+import Input from '../SearchInput/SearchInput';
+import Button from '../SearchButton/SearchButton';
 import Dropdown from '../Dropdown';
 import View from '../View/View';
+import {
+  RangeList,
+  RangeOption,
+  SearchBarProps,
+  SearchBarFieldState,
+  SearchBarValidationState
+} from '../../model';
+import { useIntl } from 'react-intl';
+import validateAddress from '../SearchBar/SearchBarInputValidator';
+
+import './searchbar.scss';
+
+const RANGES: Array<number> = [1, 5, 10, 15, 25, 35, 50];
+
+export function SearchBar({ onSubmit }: SearchBarProps) {
+  const intl = useIntl();
+  const btnSearchLabel = intl.formatMessage({ id: 'SearchBar.button.search.cta' });
+  const locationSearchPlaceholder = intl.formatMessage({
+    id: 'SearchBar.input.location.placeholder'
+  });
+
+  const rangeOptions: RangeList = RANGES.map(value => ({
+    value: value,
+    label: intl.formatMessage({ id: 'SearchBar.dropdown.range.label' }, { range: value })
+  }));
+
+  const DEFAULT_SETTINGS = {
+    location: '',
+    range: rangeOptions[0]
+  };
+
+  const [searchParamsStatus, setSearchParams] = React.useState<SearchBarFieldState>(
+    DEFAULT_SETTINGS
+  );
+  const [{ locationFieldError }, setValidation] = React.useState<SearchBarValidationState>({
+    locationFieldError: false
+  });
+
+  const updateLocation = (input: string) => {
+    setValidation({ locationFieldError: false });
+    setSearchParams({
+      ...searchParamsStatus,
+      location: input
+    });
+  };
+
+  const updateRange = (rangeInput: RangeOption) =>
+    setSearchParams({
+      ...searchParamsStatus,
+      range: rangeInput
+    });
+
+  const showValidationError = () => {
+    setValidation({ locationFieldError: true });
+    setSearchParams(DEFAULT_SETTINGS);
+  };
 
-const rangeOptions = [
-  { value: 1, label: '1 km' },
-  { value: 5, label: '5 km' },
-  { value: 10, label: '10 km' },
-  { value: 15, label: '15 km' },
-  { value: 25, label: '25 km' },
-  { value: 35, label: '35 km' },
-  { value: 50, label: '50 km' }
-];
-
-export function SearchBar() {
   return (
-    <View hAlignContent="center" vAlignContent="center" basis={300} className="search-bar">
-      <form>
-        <View vAlignContent="center">
-          <Input label="" placeholder="" value="" onChange={() => {}} />
-
-          <Dropdown
-            options={rangeOptions}
-            value={{ value: 1, label: '1 km' }}
-            onChange={() => {}}
-          />
-
-          <Button
-            label="Search"
-            flat
-            onClick={() => {
-              /*TODO : Handle Range changes*/
-            }}
-            style={{ margin: 10, width: 100 }}
-          />
-        </View>
-      </form>
+    <View vAlignContent="center" basis={300} className="search-bar">
+      <Input
+        className={`form-field location-input ${locationFieldError ? 'error-focus' : ''}`}
```

Comment:
you can use `classnames` lib to organize your classNames better:
```suggestion
        className={cx("form-field", "location-input", { "error-focus": locationFieldError })}
```

```